app/uk/gov/hmrc/cgtpropertydisposals/models/TaxEnrolmentRequest.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.models
+
+import java.time.LocalDateTime
+
+import play.api.libs.json.{Json, OFormat}
+
+final case class TaxEnrolmentRequest(
+  userId: String,
+  cgtReference: String,
+  address: Address,
+  status: String                    = "InProgress",
```

Comment:
if this were a sealed trait we'd know at a glance exactly what the different allowed statuses are

Reply:
in `getAllNonFailedEnrolmentRequests` in `DefaultTaxEnrolmentRetryRepository`, the status that's looked for is `"Retry"` - is this different than `"InProgress"`?

Reply:
Good spot - Retry should not be there anymore. There are only two statuses now - InProgress and Failed. Yes, I do need to represent this as a sealed trait.